pkg/apis/numaplane/v1alpha1/status_types.go

```diff
@@ -71,6 +71,9 @@ type Status struct {
 
 	// ObservedGeneration stores the generation value observed when setting the current Phase
 	ObservedGeneration int64 `json:"observedGeneration,omitempty"`
+
+	// UpgradeInProgress indicates the upgrade strategy currently being used and affecting the resource state or empty if no upgrade is in progress
+	UpgradeInProgress UpgradeStrategy `json:"upgradeInProgress,omitempty"`
```

Comment:
Good idea. It does seem now that we will use this across all 4 Rollouts

Reply:
Thanks. I thought to move some functions and fields I saw used by multiple controllers (ex: IsHealthy func). Then a single controller can decide to use a field/func or not.